broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/steps/BackupApiRequestHandlerStepTest.java

```diff
@@ -0,0 +1,100 @@
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.steps;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.atomix.raft.RaftServer.Role;
+import io.camunda.zeebe.broker.system.monitoring.DiskSpaceUsageMonitor;
+import io.camunda.zeebe.broker.system.partitions.TestPartitionTransitionContext;
+import io.camunda.zeebe.broker.transport.backupapi.BackupApiRequestHandler;
+import io.camunda.zeebe.logstreams.log.LogStream;
+import io.camunda.zeebe.logstreams.log.LogStreamRecordWriter;
+import io.camunda.zeebe.scheduler.ActorSchedulingService;
+import io.camunda.zeebe.scheduler.testing.TestActorFuture;
+import io.camunda.zeebe.scheduler.testing.TestConcurrencyControl;
+import io.camunda.zeebe.transport.impl.AtomixServerTransport;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+final class BackupApiRequestHandlerStepTest {
+
+  @Mock AtomixServerTransport serverTransport;
+
+  @Mock LogStream logStream;
+  @Mock LogStreamRecordWriter logStreamRecordWriter;
+
+  @Mock DiskSpaceUsageMonitor diskSpaceUsageMonitor;
+
+  @Mock ActorSchedulingService actorSchedulingService;
+
+  private final TestPartitionTransitionContext transitionContext =
+      new TestPartitionTransitionContext();
+  private BackupApiRequestHandlerStep step;
+
+  @Mock private BackupApiRequestHandler requestHandlerFromPrevRole;
+
+  @BeforeEach
+  void setup() {
+    transitionContext.setLogStream(logStream);
+    transitionContext.setGatewayBrokerTransport(serverTransport);
+    transitionContext.setConcurrencyControl(new TestConcurrencyControl());
+    transitionContext.setDiskSpaceUsageMonitor(diskSpaceUsageMonitor);
+    transitionContext.setActorSchedulingService(actorSchedulingService);
+
+    step = new BackupApiRequestHandlerStep();
+  }
+
+  @Test
+  void shouldInstallBackupRequestHandler() {
+    // given
+    when(logStream.newLogStreamRecordWriter())
+        .thenReturn(TestActorFuture.completedFuture(logStreamRecordWriter));
+    when(actorSchedulingService.submitActor(any()))
+        .thenReturn(TestActorFuture.completedFuture(null));
+
+    // when
+    transitionTo(Role.LEADER);
+
+    // then
+    assertThat(transitionContext.getBackupApiRequestHandler()).isNotNull();
+  }
+
+  @ParameterizedTest
+  @EnumSource(
+      value = Role.class,
+      names = {"FOLLOWER", "CANDIDATE", "INACTIVE"})
+  void shouldCloseBackupRequestHandler(final Role targetRole) {
```

Comment:
```suggestion
  void shouldCloseBackupRequestHandler(final Role targetRole, @Mock final BackupApiRequestHandler requestHandlerFromPrevRole) {
```

because `requestHandlerFromPrevRole` is only used here